server/controllers/airTableController.js

```diff
@@ -0,0 +1,46 @@
+import regeneratorRuntime from "regenerator-runtime";
+import request from 'request';
+require('dotenv').config();
+const Airtable = require('airtable');
+const airtableBaseAssets = 'appnvQb0LqM1nKTTQ';
+const base = new Airtable({apiKey: process.env.AIRTABLE_KEY}).base(airtableBaseAssets);
+
+const getIdAndAssetIdsOfAssetName = async (assetName) => {
+  let records = await base('Imported table').select();
+  records = await records.firstPage();
+  for(let i = 0; i < records.length; i++){
+    if(records[i].get('Asset') === assetName){
```

Comment:
I really hate `for` loops, they create a counter variable for something we can do with a `map`. You are trying to filter an element, and return when you do so an object based on it, which can be done with a `filter`. This function doesn't return anything if that element isn't found, and you also created a dynamic variable just to rewrite it instead of using two static ones.

```
const getIdAndAssetIdsOfAssetName = async (assetName) => {
  const importedRecords = await base('Imported table').select();
  const firstPageRecords = await records.firstPage();
  const selectedRecord = firstPageRecords.filter( record => record[i].get('Asset') === assetName )
  return selectedRecord ? 
    { id: selectedRecord[i].id, assetId: records[i].get('Asset IDs') } :
    { id: -1, assetId: -1 }
}
```

Reply:
Yea this one is tricky. If that function doesn't return anything it means that between the user attempting to list an asset and that function call someone removed the asset from Airtable, which supposedly is a no no, since once an asset is added to Airtable it should not be removed. By asset I mean the declaration that a new asset type exists in the platform, not to be confused with assetID.

I'll change it to a filter for readability purposes. Although for loops are still the king of performance, I like them \o/

I'll change the dynamic variables too, good point.

Reply:
I see, I just think a function should return the same type of object. Otherwise, things get tricky. This will be a requirement when using for instance TypeScript, defining each value type the function returns.

I don't know what you are talking about with performance and `for` loops haha still `O(n)` if you go through all the array, am I missing something? 😄